From 8fed12fb31b9437934525a0389e5b4a45be56bd2 Mon Sep 17 00:00:00 2001
From: Matt DeVillier <matt.devillier@puri.sm>
Date: Tue, 15 Jun 2021 10:50:17 -0500
Subject: [PATCH 2/4] mb/purism/librem_mini: Enable acoustic noise mitigation

Enable FastPkgCRampDisable for all domains, set SlowSlewRate
to fast/16 for all domains.

Test: boot Librem Mini v2, observe high frequency acoustic noise reduced.

Change-Id: I10bc2a3e6b631b8c0b430e204f376aa9a81ac683
Signed-off-by: Matt DeVillier <matt.devillier@puri.sm>
---
 .../librem_cnl/variants/librem_mini/devicetree.cb      | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/mainboard/purism/librem_cnl/variants/librem_mini/devicetree.cb b/src/mainboard/purism/librem_cnl/variants/librem_mini/devicetree.cb
index 69d1b423bd..836f4aceaf 100644
--- a/src/mainboard/purism/librem_cnl/variants/librem_mini/devicetree.cb
+++ b/src/mainboard/purism/librem_cnl/variants/librem_mini/devicetree.cb
@@ -22,6 +22,16 @@ chip soc/intel/cannonlake
 	# Misc
 	register "AcousticNoiseMitigation" = "1"
 
+	register "FastPkgCRampDisableIa" = "1"
+	register "FastPkgCRampDisableGt" = "1"
+	register "FastPkgCRampDisableSa" = "1"
+	register "FastPkgCRampDisableFivr" = "1"
+
+	register "SlowSlewRateForIa" = "3"	# fast/16
+	register "SlowSlewRateForGt" = "3"	# fast/16
+	register "SlowSlewRateForSa" = "3"	# fast/16
+	register "SlowSlewRateForFivr" = "3"	# fast/16
+
 	# Power
 	register "PchPmSlpS3MinAssert" = "3"	# 50ms
 	register "PchPmSlpS4MinAssert" = "1"	# 1s
-- 
2.30.2

